### PR TITLE
Make Iroha container configurable via WithIroha && Implement queries which are not yet presented in SDK

### PR DESCRIPTION
**Pull request by arndey**

### CODE REVIEW

File: modules/client/src/test/kotlin/jp/co/soramitsu/iroha2/InstructionsTest.kt [hunk 1]
```diff
@@ -333,6 +335,18 @@ class InstructionsTest : IrohaTest<Iroha2Client>(
         assertEquals(asset, assetsByDomainName.first())
     }
 
+    @Test
+    @WithIroha(
+        [DefaultGenesis::class],
+        configs = ["WSV_ACCOUNT_METADATA_LIMITS$IROHA_CONFIG_DELIMITER{\"max_entry_byte_size\": 65536, \"max_len\": 1048576}"]
```
Comment (Mingela) [inline modules/client/src/test/kotlin/jp/co/soramitsu/iroha2/InstructionsTest.kt:341]: how about representing that as a map or tuples, i.e. `WSV_ACCOUNT_METADATA_LIMITS to <>`

File: modules/test-tools/src/main/kotlin/jp/co/soramitsu/iroha2/testengine/IrohaRunnerExtension.kt [hunk 2]
```diff
@@ -158,6 +158,11 @@ class IrohaRunnerExtension : InvocationInterceptor, BeforeEachCallback {
                     keyPair = keyPairs[n]
                     trustedPeers = peerIds
                     ports = portsList[n]
+                    envs = withIroha.configs.associate { config ->
+                        config.split(IROHA_CONFIG_DELIMITER).let {
+                            it.first() to it.last()
```
Comment (Mingela) [inline modules/test-tools/src/main/kotlin/jp/co/soramitsu/iroha2/testengine/IrohaRunnerExtension.kt:163]: and it will make this not needed

**Review by Mingela — APPROVED** [review]

File: modules/client/src/main/kotlin/jp/co/soramitsu/iroha2/query/Queries.kt [hunk 3]
```diff
@@ -41,297 +47,263 @@ import jp.co.soramitsu.iroha2.generated.datamodel.query.transaction.FindTransact
...
 import jp.co.soramitsu.iroha2.generated.datamodel.trigger.TriggerId
 
 /**
- * Queries are sent to an Iroha peer and prompt a response with details from the current world state view.
+ * Queries are sent to an Iroha peer and prompt a response with details from the current world state view
  */
 object Queries {
 
     /**
-     * Find all accounts registered globally in the blockchain.
```
[the remark this answers is not in the document]
Comment (outoftardis) [reply modules/client/src/main/kotlin/jp/co/soramitsu/iroha2/query/Queries.kt:53]: This one can easily be omitted (as well as in other similar function descriptions) but on the line 55 it's better to keep the `.` at the end of the sentence.

File: modules/client/src/main/kotlin/jp/co/soramitsu/iroha2/query/Queries.kt [hunk 4]
```diff
@@ -41,297 +47,263 @@ import jp.co.soramitsu.iroha2.generated.datamodel.query.transaction.FindTransact
...
-    fun findAllAssets(): QueryBox.FindAllAssets {
-        return QueryBox.FindAllAssets(FindAllAssets())
-    }
+    fun findAccountsWithAsset(definitionId: AssetDefinitionId) = QueryBox.FindAccountsWithAsset(
+        FindAccountsWithAsset(definitionId.evaluatesTo())
+    )
 
     /**
-     * Return the aggregated data about the [AssetId] usage across the network.
+     * Return all known assets by value
```
Comment (outoftardis) [inline modules/client/src/main/kotlin/jp/co/soramitsu/iroha2/query/Queries.kt:96]: what about
```suggestion
     * Return the values of all known assets
```
?

File: modules/client/src/main/kotlin/jp/co/soramitsu/iroha2/query/Queries.kt [hunk 5]
```diff
@@ -41,297 +47,263 @@ import jp.co.soramitsu.iroha2.generated.datamodel.query.transaction.FindTransact
...
-    fun findAllDomains(): QueryBox.FindAllDomains {
-        return QueryBox.FindAllDomains(FindAllDomains())
-    }
+    fun findAssetDefinitionById(definitionId: AssetDefinitionId) = QueryBox.FindAssetDefinitionById(
+        FindAssetDefinitionById(definitionId.evaluatesTo())
+    )
 
     /**
-     * Return the domain corresponding to the given [DomainId].
+     * Return all the known registered domains
```
Comment (outoftardis) [inline modules/client/src/main/kotlin/jp/co/soramitsu/iroha2/query/Queries.kt:194]:
```suggestion
     * Return all known registered domains
```

File: modules/client/src/main/kotlin/jp/co/soramitsu/iroha2/query/Queries.kt [hunk 6]
```diff
@@ -41,297 +47,263 @@ import jp.co.soramitsu.iroha2.generated.datamodel.query.transaction.FindTransact
...
+     * Return the domain corresponding to the given [DomainId]
      */
-    fun findAllPeers(): QueryBox.FindAllPeers {
-        return QueryBox.FindAllPeers(FindAllPeers())
-    }
+    fun findDomainById(domainId: DomainId) = QueryBox.FindDomainById(FindDomainById(domainId.evaluatesTo()))
 
     /**
-     * Return the full set of transactions that an account with [AccountId] has submitted throughout the existence of the blockchain.
+     * Return all known peers identified by their key and accompanied by the address of the API endpoint of each
```
Comment (outoftardis) [inline modules/client/src/main/kotlin/jp/co/soramitsu/iroha2/query/Queries.kt:204]:
```suggestion
     * Return all known peers identified by their key and accompanied by the address of their API endpoint
```

File: modules/client/src/main/kotlin/jp/co/soramitsu/iroha2/query/Queries.kt [hunk 7]
```diff
@@ -41,297 +47,263 @@ import jp.co.soramitsu.iroha2.generated.datamodel.query.transaction.FindTransact
...
+     */
+    fun findTriggerById(id: TriggerId) = QueryBox.FindTriggerById(FindTriggerById(id.evaluatesTo()))
 
-    fun findAllBlocks(): QueryBox.FindAllBlocks {
-        return QueryBox.FindAllBlocks(
-            FindAllBlocks()
-        )
-    }
+    /**
+     * Return all triggers that are attached to the given [DomainId]
```
Comment (outoftardis) [inline modules/client/src/main/kotlin/jp/co/soramitsu/iroha2/query/Queries.kt:276]:
```suggestion
     * Return all the triggers that are attached to the given [DomainId]
```

File: modules/client/src/main/kotlin/jp/co/soramitsu/iroha2/query/Queries.kt [hunk 8]
```diff
@@ -41,297 +47,263 @@ import jp.co.soramitsu.iroha2.generated.datamodel.query.transaction.FindTransact
...
-                id.evaluatesTo(),
-                key.evaluatesTo()
-            )
-        )
-    }
+    fun findRoleByRoleId(roleId: RoleId) = QueryBox.FindRoleByRoleId(FindRoleByRoleId(roleId.evaluatesTo()))
 
     /**
-     * Find the value corresponding to the given key [Name] in the metadata of the trigger with the given [TriggerId].
+     * Return all the role IDs
```
Comment (outoftardis) [inline modules/client/src/main/kotlin/jp/co/soramitsu/iroha2/query/Queries.kt:252]:
```suggestion
     * Return all role IDs
```

File: modules/client/src/main/kotlin/jp/co/soramitsu/iroha2/query/Queries.kt [hunk 9]
```diff
@@ -41,297 +47,263 @@ import jp.co.soramitsu.iroha2.generated.datamodel.query.transaction.FindTransact
...
+     * Return all roles
      */
-    fun findAllRoleIds(): QueryBox.FindAllRoleIds {
-        return QueryBox.FindAllRoleIds(FindAllRoleIds())
-    }
+    fun findAllRoles() = QueryBox.FindAllRoles(FindAllRoles())
 
     /**
-     * Return the value keyed by the given [Name] in the domain corresponding to the given [DomainId].
+     * Return role by [RoleId]
```
Comment (outoftardis) [inline modules/client/src/main/kotlin/jp/co/soramitsu/iroha2/query/Queries.kt:247]:
```suggestion
     * Return the role by [RoleId]
```

File: modules/client/src/main/kotlin/jp/co/soramitsu/iroha2/query/Queries.kt [hunk 10]
```diff
@@ -41,297 +47,263 @@ import jp.co.soramitsu.iroha2.generated.datamodel.query.transaction.FindTransact
...
-            FindTransactionByHash(hash.evaluatesTo())
-        )
-    }
+    fun findPermissionTokensByAccountId(accountId: AccountId) = QueryBox.FindPermissionTokensByAccountId(
+        FindPermissionTokensByAccountId(accountId.evaluatesTo())
+    )
 
     /**
-     * Return all the role IDs that are attached to the given [AccountId].
+     * Return all the permission token definitions
```
Comment (outoftardis) [inline modules/client/src/main/kotlin/jp/co/soramitsu/iroha2/query/Queries.kt:223]:
```suggestion
     * Return all permission token definitions
```

File: modules/client/src/main/kotlin/jp/co/soramitsu/iroha2/query/Queries.kt [hunk 11]
```diff
@@ -41,297 +47,263 @@ import jp.co.soramitsu.iroha2.generated.datamodel.query.transaction.FindTransact
...
+     * Return the aggregated data about the [AssetId] usage across the network
      */
-    fun findAssetsByDomainId(domainId: DomainId): QueryBox.FindAssetsByDomainId {
-        return QueryBox.FindAssetsByDomainId(FindAssetsByDomainId(domainId.evaluatesTo()))
-    }
+    fun findAssetById(assetId: AssetId) = QueryBox.FindAssetById(FindAssetById(assetId.evaluatesTo()))
 
     /**
-     * Search the network for all assets that match the given [Name].
+     * Return all assets that are registered in the given domain [DomainId]
```
Comment (outoftardis) [inline modules/client/src/main/kotlin/jp/co/soramitsu/iroha2/query/Queries.kt:106]:
```suggestion
     * Return all the assets that are registered in the given domain [DomainId]
```

File: modules/client/src/main/kotlin/jp/co/soramitsu/iroha2/query/Queries.kt [hunk 12]
```diff
@@ -41,297 +47,263 @@ import jp.co.soramitsu.iroha2.generated.datamodel.query.transaction.FindTransact
...
+     * Return all the accounts that have the given [Name]
      */
-    fun findAccountsByName(name: Name): QueryBox.FindAccountsByName {
-        return QueryBox.FindAccountsByName(FindAccountsByName(name.evaluatesTo()))
-    }
+    fun findAccountsByName(name: Name) = QueryBox.FindAccountsByName(FindAccountsByName(name.evaluatesTo()))
 
     /**
-     * Return all accounts that belong to a specific domain [DomainId].
+     * Return all accounts that belong to a specific domain [DomainId]
```
Comment (outoftardis) [inline modules/client/src/main/kotlin/jp/co/soramitsu/iroha2/query/Queries.kt:82]:
```suggestion
     * Return all the accounts that belong to a specific domain [DomainId]
```

File: modules/client/src/main/kotlin/jp/co/soramitsu/iroha2/query/Queries.kt [hunk 13]
```diff
@@ -41,297 +47,263 @@ import jp.co.soramitsu.iroha2.generated.datamodel.query.transaction.FindTransact
...
-            FindAccountsByDomainId(domainId.evaluatesTo())
-        )
-    }
+    fun findAccountsByDomainId(domainId: DomainId) = QueryBox.FindAccountsByDomainId(
+        FindAccountsByDomainId(domainId.evaluatesTo())
+    )
 
     /**
-     * Return all known assets by value.
+     * Return all accounts that belong to a specific asset definition [AssetDefinitionId]
```
Comment (outoftardis) [inline modules/client/src/main/kotlin/jp/co/soramitsu/iroha2/query/Queries.kt:89]:
```suggestion
     * Return all the accounts that belong to a specific asset definition [AssetDefinitionId]
```

**Review by outoftardis — APPROVED** [review]

**Pull request merged**